src/Compilers/VisualBasic/Test/Syntax/Parser/ParserRegressionTests.vb

```diff
@@ -934,4 +934,17 @@ End Enum
         Assert.Equal(Syntax.InternalSyntax.Scanner.BadTokenCountLimit, tree.GetDiagnostics().Where(Function(d) d.Code = ERRID.ERR_IllegalChar).Count())
     End Sub
 
+    <Fact, WorkItem(48587, "https://github.com/dotnet/roslyn/issues/48587")>
+    Public Sub ParseTrailingTextAfterPropertyWithParentheses()
+        ParseAndVerify(<![CDATA[
+Class C
+    Public ReadOnly Property NumberOfResult1() String Integer JohnDoe WwwIIWww Wow
+    Public ReadOnly Property NumberOfResult2() Some unexpected tokens As Integer
```

Comment:
>Some unexpected tokens As Integer [](start = 47, length = 33)

Please also test with a comment at the end of the line, line continuation and line continuation followed by a comment.